x/tokenomics/keeper/token_logic_modules.go

```diff
@@ -285,22 +316,9 @@ func (k Keeper) TokenLogicModuleRelayBurnEqualsMint(
 	}
 	logger.Info(fmt.Sprintf("sent (%v) from the supplier module to the supplier account with address %q", settlementCoin, supplier.OperatorAddress))
 
-	// TODO_MAINNET: Decide on the behaviour here when an app is over serviced.
-	// If an app has 10 POKT staked, but the supplier earned 20 POKT. We still
-	// end up minting 20 POKT but only burn 10 POKT from the app. There are
-	// questions and nuance here that needs to be addressed.
-
-	// Verify that the application has enough uPOKT to pay for the services it consumed
-	if application.GetStake().IsLT(settlementCoin) {
-		settlementCoin, err = k.handleOverservicedApplication(ctx, application, settlementCoin)
-		if err != nil {
-			return err
-		}
-	}
-
 	// Burn uPOKT from the application module account which was held in escrow
 	// on behalf of the application account.
-	if err = k.bankKeeper.BurnCoins(
+	if err := k.bankKeeper.BurnCoins(
 		ctx, apptypes.ModuleName, sdk.NewCoins(settlementCoin),
 	); err != nil {
 		return tokenomicstypes.ErrTokenomicsApplicationModuleBurn.Wrapf("burning %s from the application module account: %v", settlementCoin, err)
```

Comment:
I'm noticing (in the lines below: 329-333) that by the time we determine whether or not the application has enough stake to decrement, we've already burnt their tokens from escrow (app module acct.). Since we return an error in the case where app stake is insufficient and leave the stake unchanged, the app will continue to have stake representing tokens in escrow which have already been burnt (and then some).

Wdyt about updating the application stake before burning from the app module acct.?